tests/testutils/kubeutils/kind_cluster.go

```diff
@@ -0,0 +1,180 @@
+// Copyright Splunk, Inc.
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kubeutils
+
+import (
+	"bytes"
+	"context"
+	"crypto/sha256"
+	"fmt"
+	"os"
+	"strings"
+	"text/template"
+	"time"
+
+	"github.com/docker/docker/api/types"
+	docker "github.com/docker/docker/client"
+
+	"github.com/stretchr/testify/require"
+	"go.uber.org/multierr"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	kubectlcmd "k8s.io/kubectl/pkg/cmd"
+	kubectlcmdutil "k8s.io/kubectl/pkg/cmd/util"
+	kindcmd "sigs.k8s.io/kind/pkg/cmd"
+	"sigs.k8s.io/kind/pkg/cmd/kind"
+
+	"github.com/signalfx/splunk-otel-collector/tests/testutils"
+)
+
+const kindConfigTemplate = `kind: Cluster
+apiVersion: kind.x-k8s.io/v1alpha4
+nodes:
+  - role: control-plane
+    {{- if .ExposedPorts }}
+    extraPortMappings:
+    {{- range $hostPort, $containerPort := .ExposedPorts }}
+      - containerPort: {{ $containerPort }}
+        hostPort: {{ $hostPort }}
+        listenAddress: "0.0.0.0"
+        protocol: tcp
+    {{- end }}
+    {{- end }}
+`
+
+type KindCluster struct {
+	Testcase     *testutils.Testcase
+	Clientset    *kubernetes.Clientset
+	ExposedPorts map[uint16]uint16
+	Name         string
+	Kubeconfig   string
+	Config       string
+}
+
+func NewKindCluster(t *testutils.Testcase) *KindCluster {
+	t.TB.TempDir()
```

Comment:
I think this is not necessary, please advise?

Reply:
Was testing this helper but opted not to require deletion on cleanup.